### PR TITLE
Config version aware - Case Working

### PR DESCRIPTION
**Pull request by mkrad7**

### CODE REVIEW

File: src/cases/use-cases/find-cases.use-case.js
Comment (justin-esynergy) [inline src/cases/use-cases/find-cases.use-case.js:101]: I think this filter loses the version-aware part of the access check.

`findWorkflowsUseCase(roleFilter)` checks roles against workflow documents, but there can now be multiple documents with the same `code` and different `version` / `requiredRoles`. Once we map those workflows down to just `workflowCode`, any matching version makes the whole code visible.

Example: if `pigs-might-fly@1.0.0` allows `ROLE_A`, but a case now resolves to `pigs-might-fly@1.1.0` which requires `ROLE_B`, a `ROLE_A` user can still get `pigs-might-fly` into this query and see that case in the list.

Can we keep the auth check tied to the resolved workflow version?

Maybe worth a chat with Martin to see if we need to worry about this edge case, maybe we would never change required roles on a workflow.

File: src/cases/use-cases/resolve-and-fetch-workflow.use-case.js [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+  );
+};
+
+const guardFetchStatus = async (
+  configVersion,
+  workflowCode,
+  resolvedVersion,
+) => {
+  if (
+    configVersion.fetchAttempts >= MAX_FETCH_ATTEMPTS &&
```
Comment (justin-esynergy) [inline src/cases/use-cases/resolve-and-fetch-workflow.use-case.js]: I think this guard can mark a successfully fetched config version as permanently broken.

`updateFetchStatus(..., FetchStatus.Fetched)` increments `fetchAttempts`, so a version can end up with `fetchStatus: "fetched"` and `fetchAttempts: 5`. On the next resolve, this check runs before we try `findStoredWorkflow`, and because it only excludes `PermanentError`, it changes that fetched version to `permanent_error`.

Can we either skip this max-attempt check when `fetchStatus` is `Fetched`, or check the cached workflow before applying the retry limit? A test for "fetched with 5 attempts still resolves from cache" would cover it.

File: src/cases/use-cases/find-cases.use-case.js
Comment (mkrad7) [reply src/cases/use-cases/find-cases.use-case.js:101]: Valid edge case. It only is valid if requiredRoles changes between versions of the same workflow code, which the current process doesn't do — roles are treated as a property of the workflow code, not of a version IMHO

I would recommend we keep this PR and take your suggestion to confirm with Martin whether per-version role changes need to be supported. If they do, we'll raise a follow-up ticket for a version-aware access check.

File: src/cases/use-cases/resolve-and-fetch-workflow.use-case.js (hunk 1 above)
Comment (mkrad7) [reply src/cases/use-cases/resolve-and-fetch-workflow.use-case.js]: Fixed, on both fronts: updateFetchStatus no longer increments fetchAttempts when recording Fetched (matching GAS), and guardFetchStatus now skips the max-attempts escalation when the status is already Fetched

so even rows that accumulated attempts before succeeding can't be flipped to permanent_error. /Also added the test you suggested ("fetched with 5 attempts still resolves from cache") plus a repository test asserting no $inc on successful fetches.

File: src/cases/use-cases/find-cases.use-case.js
Comment (justin-esynergy) [reply src/cases/use-cases/find-cases.use-case.js:101]: Ok fair enough, it is an edge case, can we document it some where so it gets discussed with Martin at some point.

File: src/cases/repositories/workflow.repository.js [hunk 2]
```diff
@@ -191,11 +191,21 @@ export const findAll = async (query) => {
...
 
-export const findByCode = async (code, version = "0.0.0") => {
-  const workflowDocument = await db.collection(collection).findOne({
-    code,
-    version,
-  });
+export const findByCode = async (code) => {
+  const workflowDocument = await db
+    .collection(collection)
+    .findOne({ code }, { sort: { version: -1 } });
```
Comment (justin-esynergy) [inline src/cases/repositories/workflow.repository.js]: Is "latest version" lookup sorting version as a string?

That works for `1.0.3` vs `1.0.2`, but it breaks once a version part has different digit lengths. For example, Mongo string sorting will put `1.9.0` ahead of `1.10.0`, even though `1.10.0` is the newer version?

Comment (mkrad7) [reply src/cases/repositories/workflow.repository.js]: findByCode(code) should return the legacy 0.0.0 workflow rather than "latest". Changed it to findByCode(code, version = "0.0.0") with an exact match (mirroring GAS), which removes the string sort entirely. 

Latest-version resolution already goes through config_versions.findLatestForMajor, which sorts on the numeric major/minor/patch fields, so 1.9.0 vs 1.10.0 ordering is handled correctly there. 

Tests updated to cover the default and explicit-version lookups.

**Review by justin-esynergy — APPROVED** [review]

**Review by justin-esynergy — APPROVED** [review]
